Add changelog and changelog verifier (#111)
opensearch-project/opensearch-net: CONTRIBUTING.md

You may type this line on your own when writing your commit messages. However, if your user.name and user.email are set in your git configs, you can use `-s` or `– – signoff` to add the `Signed-off-by` line to the end of the commit message.

## Changelog

> OpenSearch-Net maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.

Reviewer: ```suggestion
OpenSearch-Net maintains a version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.
```

> Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.

Reviewer: ```suggestion
Briefly, the changes are curated by version, with the changes to the main branch added chronologically to the `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
```

> ### How to add my changes to [CHANGELOG](CHANGELOG.md)?

Reviewer: If you want to frame this as a question from the contributors perspective I'd go with:
```suggestion
### How do I add my changes to the [CHANGELOG](CHANGELOG.md)?
```

Reviewer: Otherwise if you want it as a statement of us telling the contributor it'd be more like:
```suggestion
### How to add your changes to the [CHANGELOG](CHANGELOG.md)
```

> As a contributor, you must ensure that every pull request has the changes listed out within the corresponding version and appropriate section of [CHANGELOG](CHANGELOG.md) file.
> 
> Adding in the change is two step process -

Reviewer: ```suggestion
As a contributor, you must ensure that every pull request has its changes listed out within the corresponding version and appropriate section of the [CHANGELOG](CHANGELOG.md) file.

Adding in the change is a two step process -
```
